detection_rules/semver.py

```diff
@@ -4,14 +4,21 @@
 
 """Helper functionality for comparing semantic versions."""
 import re
+from typing import Iterable, Optional, Union
 
 
 class Version(tuple):
 
-    def __new__(cls, version):
+    def __new__(cls, version: Union[Iterable, str], pad: Optional[int] = None) -> 'Version':
```

Comment:
Can we remove pad if we don't need it any more?
I think it makes more sense to truncate somewhere else than to pad. You also don't know what type to pad it with. integer? string?

Reply:
```python
a = Version("1.0.2")
b = Version("1.0")

assert a > b
assert a[:len(b)] == b
```

Reply:
I'll remove it 👍